swiftpkg/internal/registry_configs.bzl

```diff
@@ -0,0 +1,34 @@
+"""Module for resolving registry URLs from scopes."""
+
+def _read(repository_ctx):
+    registry_config_json = repository_ctx.read(
+        Label("@//:.swiftpm/configuration/registries.json"),
```

Comment:
We can add an attribute defined as `attr.label(allow_single_file = True)` to `swift_registry_package()`. We can read the value in `_swift_registry_package_impl()` and pass the path to this function.

Reply:
OK, I created a `registry_config` attribute for this. It still defaults to `Label("@//:.swiftpm/configuration/registries.json")`, rather than emitting that to the generated rules, though. Is that OK? I'm worried that letting this be customized is a bit dangerous, because putting any other value in here means it won't be using the same config that `swift package resolve` does.

I think we might want to also pass a `--config-path` arg to `swift` to prevent it from using global config outside the workspace.

Reply:
Sorry for the delay. I had many family obligations over the past 2 weeks. 

Regarding the label default, that is fine. In another comment, I suggest that we update the doc for the attribute stating where the file comes from / how it is typically generated.

>I think we might want to also pass a --config-path arg to swift to prevent it from using global config outside the workspace.

Where do you want to add that?

Reply:
> > I think we might want to also pass a --config-path arg to swift to prevent it from using global config outside the workspace.
> 
> Where do you want to add that?

In the `SwiftBin` wrapper that calls `swift package resolve`. By passing a `--config-path` to that command, we can make `swift` only look at the config in this workspace (which is the only one RSPM is looking at). So the command would become something like:
```sh
swift package resolve --config-path .swiftpm/configuration
```

`--config-path` sets the location of the _shared_ config, which is usually `~/.swiftpm/configuration`. The idea here is just to make it identical to the project config so that there's only one config to consider. Setting it to something like `/dev/null` would probably have the same effect (but is maybe more fragile if a new version of swift decides the directly must exist).

Reply:
OK. I am fine with having the new parameter passing the `--config-path` value. However, as I may have mentioned in another thread, the `swift_update_packages` macro may disappear before 1.0. I will open a discussion about that shortly.